### PR TITLE
Vol 2735: Durations should be specified as type `time.Duration` not `int`

### PR DESCRIPTION
**Pull request by nsharma70**

### CODE REVIEW

File: pkg/db/kvstore/consulclient.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ type ConsulClient struct {
 }
 
 // NewConsulClient returns a new client for the Consul KV store
-func NewConsulClient(addr string, timeout int) (*ConsulClient, error) {
+func NewConsulClient(addr string, timeout time.Duration) (*ConsulClient, error) {
 
 	duration := GetDuration(timeout)
```
Comment (gkumar78) [inline pkg/db/kvstore/consulclient.go]: Call to GetDuration is not needed here

File: pkg/db/kvstore/kvutils.go [hunk 2]
```diff
@@ -20,13 +20,13 @@ import (
 	"time"
 )
 
-// GetDuration converts a timeout value from int to duration.  If the timeout value is
+// GetDuration returns the timeout value.  If the timeout value is
 // either not set of -ve then we default KV timeout (configurable) is used.
-func GetDuration(timeout int) time.Duration {
+func GetDuration(timeout time.Duration) time.Duration {
 	if timeout <= 0 {
```
Comment (gkumar78) [inline pkg/db/kvstore/kvutils.go]: Can time.Duration value can be negative? I guess this check should be present at place where time.Duration value is getting initialized from a int value

File: pkg/db/kvstore/kvutils.go [hunk 3]
```diff
@@ -20,13 +20,13 @@ import (
...
 		return defaultKVGetTimeout * time.Second
 	}
-	return time.Duration(timeout) * time.Second
```
Comment (gkumar78) [inline pkg/db/kvstore/kvutils.go:29]: Have a tested that timeout logic is working fine after these changes? Set timeout of 5 second and test

File: pkg/db/kvstore/consulclient.go [hunk 4]
```diff
@@ -76,7 +76,7 @@ func (c *ConsulClient) List(ctx context.Context, key string) (map[string]*KVPair
 	deadline, _ := ctx.Deadline()
 	kv := c.consul.KV()
 	var queryOptions consulapi.QueryOptions
-	queryOptions.WaitTime = GetDuration(deadline.Second())
```
Comment (gkumar78) [inline pkg/db/kvstore/consulclient.go:79]: Not the correct approach. When a int value representing time is converted to Duration, it should be multiplied by base value representing that time unit e.g. time.Second here

File: pkg/ponresourcemanager/ponresourcemanager.go [hunk 5]
```diff
@@ -124,7 +125,7 @@ const (
 	POOL            = "pool"
 	NUM_OF_PON_INTF = 16
 
-	KVSTORE_RETRY_TIMEOUT = 5
+	KVSTORE_RETRY_TIMEOUT = 5 * time.Second
```
Comment (gkumar78) [inline pkg/ponresourcemanager/ponresourcemanager.go]: Change to following to change data type of constant to duration

time.Duration(5) * time.Second

File: pkg/db/kvstore/kvutils.go [hunk 6]
```diff
@@ -15,19 +15,16 @@
  */
 package kvstore
 
-import (
-	"fmt"
-	"time"
-)
+import "fmt"
 
 // GetDuration converts a timeout value from int to duration.  If the timeout value is
```
Comment (gkumar78) [inline pkg/db/kvstore/kvutils.go]: Remove commented out method

File: pkg/db/kvstore/etcdclient.go [hunk 7]
```diff
@@ -160,15 +160,15 @@ func (c *EtcdClient) Delete(ctx context.Context, key string) error {
...
+func (c *EtcdClient) Reserve(ctx context.Context, key string, value interface{}, ttl time.Duration) (interface{}, error) {
 	// Validate that we can convert value to a string as etcd API expects a string
 	var val string
 	var er error
 	if val, er = ToString(value); er != nil {
 		return nil, fmt.Errorf("unexpected-type%T", value)
 	}
 
-	resp, err := c.ectdAPI.Grant(ctx, ttl)
+	resp, err := c.ectdAPI.Grant(ctx, int64(ttl))
```
Comment (gkumar78) [inline pkg/db/kvstore/etcdclient.go]: ttl is usually represented in seconds. use ttl.Seconds() method to retrieve seconds part of duration

File: pkg/db/kvstore/consulclient.go [hunk 8]
```diff
@@ -166,7 +164,7 @@ func (c *ConsulClient) deleteSession() {
 	c.session = nil
 }
 
-func (c *ConsulClient) createSession(ttl int64, retries int) (*consulapi.Session, string, error) {
+func (c *ConsulClient) createSession(ttl time.Duration, retries int) (*consulapi.Session, string, error) {
 	session := c.consul.Session()
 	entry := &consulapi.SessionEntry{
 		Behavior: consulapi.SessionBehaviorDelete,
```
Comment (gkumar78) [inline pkg/db/kvstore/consulclient.go:175]: Better use ttl coming in the argument in this call as fmt.Print(ttl).

File: pkg/db/kvstore/consulclient.go [hunk 9]
```diff
@@ -97,7 +95,7 @@ func (c *ConsulClient) Get(ctx context.Context, key string) (*KVPair, error) {
 	deadline, _ := ctx.Deadline()
 	kv := c.consul.KV()
 	var queryOptions consulapi.QueryOptions
-	queryOptions.WaitTime = GetDuration(deadline.Second())
```
Comment (gkumar78) [inline pkg/db/kvstore/consulclient.go:100]: Not the right implementation if WaitTime is the duration till when to wait for response generation

File: pkg/db/kvstore/client.go [hunk 10]
```diff
@@ -15,11 +15,14 @@
...
-import "context"
+import (
+       "context"
+       "time"
+)
 
 const (
 	// Default timeout in seconds when making a kvstore request
-	defaultKVGetTimeout = 5
+	defaultKVGetTimeout = 5 * time.Second
```
Comment (gkumar78) [inline pkg/db/kvstore/client.go]: time.Duration(5) * time.Second

File: pkg/db/kvstore/kvutils.go [hunk 11]
```diff
@@ -15,19 +15,16 @@
...
-		return defaultKVGetTimeout * time.Second
-	}
-	return time.Duration(timeout) * time.Second
-}
+//func GetDuration(timeout int) time.Duration {
+//	if timeout <= 0 {
+//		return defaultKVGetTimeout
+//	}
+//	return time.Duration(timeout) * time.Second
+//}
```
Comment (divya2504) [inline pkg/db/kvstore/kvutils.go]: remove the commented lines

File: pkg/db/kvstore/kvutils_test.go [hunk 12]
```diff
@@ -18,29 +18,29 @@ package kvstore
 import (
 	"github.com/stretchr/testify/assert"
 	"testing"
-	"time"
+//	"time"
```
Comment (divya2504) [inline pkg/db/kvstore/kvutils_test.go]: remove commented line

File: pkg/db/kvstore/kvutils_test.go [hunk 13]
```diff
@@ -18,29 +18,29 @@ package kvstore
...
-	actualResult := GetDuration(-1)
-	var expectedResult = defaultKVGetTimeout * time.Second
+//func TestDurationWithNegativeTimeout(t *testing.T) {
+//	actualResult := GetDuration(-1)
+//	var expectedResult = defaultKVGetTimeout
 
-	assert.Equal(t, expectedResult, actualResult)
-}
+//	assert.Equal(t, expectedResult, actualResult)
+//}
```
Comment (divya2504) [inline pkg/db/kvstore/kvutils_test.go]: remove commented lines from all the files

File: pkg/mocks/etcd_server_test.go [hunk 14]
```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/phayes/freeport"
 	"github.com/stretchr/testify/assert"
 	"os"
+        "time"
```
Comment (divya2504) [inline pkg/mocks/etcd_server_test.go:26]: run go fmt on all the files

File: pkg/ponresourcemanager/ponresourcemanager.go (hunk 5 above)
Comment (divya2504) [reply pkg/ponresourcemanager/ponresourcemanager.go]: Change in all the places data type of constant to duration

File: pkg/ponresourcemanager/ponresourcemanager_test.go [hunk 15]
```diff
@@ -26,6 +26,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"strings"
 	"testing"
+        "time"
```
Comment (divya2504) [inline pkg/ponresourcemanager/ponresourcemanager_test.go]: run go fmt

File: pkg/db/kvstore/kvutils.go (hunk 11 above)
Comment (divya2504) [reply pkg/db/kvstore/kvutils.go]: remove the commented lines

File: pkg/db/kvstore/consulclient.go [hunk 16]
```diff
@@ -166,11 +169,11 @@ func (c *ConsulClient) deleteSession() {
 	c.session = nil
 }
 
-func (c *ConsulClient) createSession(ttl int64, retries int) (*consulapi.Session, string, error) {
+func (c *ConsulClient) createSession(ttl time.Duration, retries int) (*consulapi.Session, string, error) {
 	session := c.consul.Session()
 	entry := &consulapi.SessionEntry{
 		Behavior: consulapi.SessionBehaviorDelete,
-		TTL:      "10s", // strconv.FormatInt(ttl, 10) + "s", // disable ttl
+		TTL:      fmt.Sprint(ttl),
```
Comment (divya2504) [inline pkg/db/kvstore/consulclient.go:176]: Provide the right formating

File: pkg/db/kvstore/consulclient.go [hunk 17]
```diff
@@ -76,7 +75,9 @@ func (c *ConsulClient) List(ctx context.Context, key string) (map[string]*KVPair
 	deadline, _ := ctx.Deadline()
 	kv := c.consul.KV()
 	var queryOptions consulapi.QueryOptions
-	queryOptions.WaitTime = GetDuration(deadline.Second())
+	// Substract current time from deadline to get the waitTime duration
+	queryOptions.WaitTime = deadline.Sub(time.Now())
```
Comment (divya2504) [inline pkg/db/kvstore/consulclient.go:79]: have u tested this?

**Pull request closed without merging**